src/ast/passes/semantic_analyser.cpp

```diff
@@ -2827,6 +2844,46 @@ void SemanticAnalyser::visit(AssignConfigVarStatement &assignment)
   Visit(assignment.expr);
 }
 
+void SemanticAnalyser::visit(VarDeclStatement &decl)
+{
+  std::string var_ident = decl.var->ident;
+  auto &assignTy = decl.type;
+  decl.var->type = assignTy;
+
+  // This should be the first time we're seeing this variable
+  auto decl_search = declared_variables_[scope_].find(var_ident);
+  if (decl_search != declared_variables_[scope_].end()) {
+    if (!areLocationsEqual(decl_search->second, decl.loc)) {
+      // Note this will change with proper block/lexical scoping
+      LOG(ERROR, decl.loc, err_)
+          << "Variable " << var_ident << " was already declared.";
+    }
+  }
+
+  auto search = variable_val_[scope_].find(var_ident);
+  bool var_found = search != variable_val_[scope_].end();
+  if (var_found) {
```

Comment:
Can be merged into a single line?
```suggestion
  if (search != variable_val_[scope_].end()) {
```